azurerm_machine_learning_compute_instance : issue on policy instance name

terraform version : i test with 0.14.6 & 0.15.5
================================================
terraform {
  required_version = ">= 0.13.0"
  required_providers {
    azurerm = ">= 2.85.0"
  }
}
================================================

hello everyone 
I created a terraform module which creates both an aml workspace, as well as a compute cluster or / and compute instances, the problem I have is when creating compute instances, an error on terraform plan :

` Error: invalid value for name (It can include letters, digits and dashes. It must start with a letter, end with a letter or digit, and be between 2 and 16 characters in length.)`


whereas when I try to create the same computed instance with the same name on the Azure portal, the naming rule is not the same

`Compute instance name must be between 3 to 24 characters long.`


my ressource on the module  
======================

```terraform
#########################################   Instance Compute AML     ######################################

resource "azurerm_machine_learning_compute_instance" "compute_instance" {
  count                         = var.create_compute_instance ? 1 : 0
  **name                          = var.compute_instance_name**
  location                      = var.location
  machine_learning_workspace_id = var.create_aml_workspace ? azurerm_machine_learning_workspace.main[0].id : var.personal_aml_workspace_id
  virtual_machine_size          = var.compute_instance_size
  subnet_resource_id            = var.compute_instance_subnet_id
  authorization_type            = "personal" # only Possible values
  tags                          = var.tags

  ssh {
    #NOTE :  the user name is always : azureuser (obligated by azureML)
    public_key = var.create_compute_instance ? module.mod-tls-sshkey.public_key_openssh : null
  }

  identity {
    type         = var.create_compute_instance ? "SystemAssigned" : null #  Possible values are SystemAssigned, UserAssigned and SystemAssigned,UserAssigned
    identity_ids = var.create_compute_instance ? var.compute_instance_identity_ids : null

  }
  assign_to_user {
    object_id = var.create_compute_instance ? var.compute_instance_assign_to_user_id : null
    tenant_id = var.create_compute_instance ? var.tenant_id : null
  }
}
```

i had a crashed plugin when i put an instance name with 16 caracters


> Error: Plugin did not respond

 ` with module.mod-azurerm-compute-std-compute-instance.azurerm_machine_learning_compute_instance.compute_instance[0],
  on .terraform/modules/mod-azurerm-compute-std-compute-instance/main.tf line 113, in resource "azurerm_machine_learning_compute_instance" "compute_instance":
 113: resource "azurerm_machine_learning_compute_instance" "compute_instance" {

The plugin encountered an error, and failed to respond to the
plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain
more details.

Stack trace from the terraform-provider-azurerm_v2.85.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 1139 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/machinelearning.expandComputePersonalComputeInstanceSetting(...)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/machinelearning/machine_learning_compute_instance_resource.go:293
github.com/hashicorp/terraform-provider-azurerm/internal/services/machinelearning.resourceComputeInstanceCreate(0xc00177da00, 0x5ee1820, 0xc0016ed880, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/machinelearning/machine_learning_compute_instance_resource.go:189 +0x1a47
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc000e467e0, 0x6fcf388, 0xc001b57980, 0xc00177da00, 0x5ee1820, 0xc0016ed880, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:318 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000e467e0, 0x6fcf388, 0xc001b57980, 0xc0019a1ad0, 0xc00177d780, 0x5ee1820, 0xc0016ed880, 0x0, 0x0, 0x0, ...)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:456 +0x67b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00000dd58, 0x6fcf388, 0xc001b57980, 0xc002c74640, 0xc001b57980, 0x641d0c0, 0xc0029abd00)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:977 +0xacf
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc001e80000, 0x6fcf430, 0xc001b57980, 0xc0003adab0, 0xc001e80000, 0xc0029abd40, 0xc0016c0ba0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x641d0c0, 0xc001e80000, 0x6fcf430, 0xc0029abd40, 0xc001ee1e60, 0x0, 0x6fcf430, 0xc0029abd40, 0xc002213900, 0x1030)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000234a80, 0x7023fd8, 0xc000fbc180, 0xc00132a000, 0xc001395470, 0xad6ece0, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000234a80, 0x7023fd8, 0xc000fbc180, 0xc00132a000, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000f9aa40, 0xc000234a80, 0x7023fd8, 0xc000fbc180, 0xc00132a000)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x1fd

Error: The terraform-provider-azurerm_v2.85.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

##[error]Error: The process '/opt/hostedtoolcache/terraform/0.15.5/x64/terraform' failed with exit code 1`

--- a/internal/services/machinelearning/machine_learning_compute_instance_resource.go
+++ b/internal/services/machinelearning/machine_learning_compute_instance_resource.go
@@ -50,7 +50,7 @@ func resourceComputeInstance() *pluginsdk.Resource {
 				ForceNew: true,
 				ValidateFunc: validation.StringMatch(
 					regexp.MustCompile(`^[a-zA-Z][a-zA-Z0-9-]{3,24}$`),
-					"It can include letters, digits and dashes. It must start with a letter, end with a letter or digit, and be between 2 and 16 characters in length."),
+					"It can include letters, digits and dashes. It must start with a letter, end with a letter or digit, and be between 3 and 24 characters in length."),
 			},
 
 			"machine_learning_workspace_id": {

--- a/internal/services/machinelearning/machine_learning_compute_instance_resource.go
+++ b/internal/services/machinelearning/machine_learning_compute_instance_resource.go
@@ -49,8 +49,8 @@ func resourceComputeInstance() *pluginsdk.Resource {
 				Required: true,
 				ForceNew: true,
 				ValidateFunc: validation.StringMatch(
-					regexp.MustCompile(`^[a-zA-Z][a-zA-Z0-9-]{2,23}$`),
-					"It can include letters, digits and dashes. It must start with a letter, end with a letter or digit, and be between 3 and 24 characters in length."),
+					regexp.MustCompile(`^[a-zA-Z][a-zA-Z0-9-]{3,24}$`),
+					"It can include letters, digits and dashes. It must start with a letter, end with a letter or digit, and be between 2 and 16 characters in length."),
 			},
 
 			"machine_learning_workspace_id": {

